superchain/superchain-version_test.go

```diff
@@ -0,0 +1,61 @@
+package superchain_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func checkErr(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestContractVersionsCheck will fail if the superchain semver file
+// is not read correctly.
+func TestContractVersionsCheck(t *testing.T) {
+	if err := superchain.SuperchainSemver.SanityCheck(); err != nil {
+		t.Fatal(err)
+	}
+
+	desiredSemver := superchain.SuperchainSemver.OptimismPortal
+	// TODO complete all members of SuperchainSemver
+	// TODO resolve missing contracts
+	// TODO resolve extraneous contracts
+	for _, chain := range superchain.OPChains {
+		client, err := ethclient.Dial(superchain.Superchains[chain.Superchain].Config.L1.PublicRPC)
+		checkErr(t, err)
+		actualSemver, err := getVersion(context.Background(), common.Address(superchain.Addresses[chain.ChainID].OptimismPortalProxy), client) // TODO resolve difference in names
+		checkErr(t, err)
+		if desiredSemver != actualSemver {
+			t.Fatalf("OptimismPortalProxy should have version %v but has version %v (%s on %s)", desiredSemver, actualSemver, chain.Name, chain.Superchain)
```

Comment:
It's not yet clear to me if we should require an exact match or allow for the desired semver to declare a range of acceptable versions https://devhints.io/semver .

Reply:
I think for now we should do `>=` [whatever OP Mainnet has] - this will allow chains like Base, which have newer versions than OP Mainnet, to pass this test. Eventually we'll want to do strict equality - but at that point, we'll hopefully be doing strict bytecode equality checking anyways